src/index.test.ts

```diff
@@ -192,3 +193,32 @@ test('Deps parser and plugin are same version', async (t) => {
   const pluginRange = ourPeerDeps['@typescript-eslint/eslint-plugin']
   t.is(parserRange.split('^')[1], pluginRange.split('>=')[1])
 })
+
+interface RulesConfig {
+  [name: string]: object | string | number | boolean | null
+}
+
+/**
+ * Test that objects' values do not hold same memory references
+ */
+function testNotHaveObjectReferences (t: ExecutionContext<unknown>, rules: RulesConfig, comparisonRules: RulesConfig, skipSameKey: boolean = false): void {
+  for (const ruleName in rules) {
+    const ruleConfig = rules[ruleName]
+    if (typeof ruleConfig !== 'object') continue // Non-objects use copy-by-value
+
+    for (const comparisonRuleName in comparisonRules) {
+      if (skipSameKey && ruleName === comparisonRuleName) continue
+      t.not(ruleConfig, comparisonRules[comparisonRuleName])
+    }
+  }
+}
+
+test('No references in rules config', (t) => {
+  for (const override of exported.overrides) {
+    const rules: RulesConfig = override.rules
```

Comment:
No need for this type, then, I guess.

Reply:
It is required due to `Element implicitly has an 'any' type because expression of type 'string' can't be used to index type` if use `standardRules[ruleName]`.